var/spack/repos/builtin/packages/swak4foam/package.py

```diff
@@ -0,0 +1,203 @@
+    depends_on('bison@:2.7',  type='build')  # swak4Foam only tested with 2.x
+
+    # General patches
+    assets = ['local-userdir.sh']
+
+    build_script = './spack-Allwmake'  # <- Generated by patch() method.
+    build_userdir = 'spack-swak4foam'  # Build user APPBIN, LIBBIN into here
+    config_file  = 'swakConfiguration'
+
+    def create_build_script(self):
+        """Generate a wrapper build script."""
+        # Only call a build stage, not patch stage since it tries to discover
+        # the foam project directory based on the existence of file
+        # Build wrapper script
+        spec = self.spec
+        foam_spec = spec['openfoam']
+        foam_projdir = find_foam_projectdir(foam_spec)  # WM_PROJECT_DIR
+        foam_instdir = os.path.dirname(foam_projdir)    # WM_PROJECT_INST_DIR
+
+        script = join_path(self.stage.source_path, self.build_script)
+        with open(script, 'w') as out:
+            out.write(
+                """#!/bin/bash
+export FOAM_INST_DIR={foam_instdir}
+export WM_PROJECT_DIR={foam_projdir}
+. $WM_PROJECT_DIR/etc/bashrc ''  # No arguments
+
+# Local build (user appbin, libbin)
+. $PWD/local-userdir.sh  {user_dir}
+
+# Force bison to be found first
+bison_bin={bison_bin}
+PATH=$bison_bin:$PATH
+
+echo Build swak4foam with SPACK for $WM_PROJECT-$WM_PROJECT_VERSION
+echo WM_PROJECT_DIR = $WM_PROJECT_DIR
+
+./Allwmake  # No arguments
+#
+"""
+                .format(
+                    foam_instdir=foam_instdir,
+                    foam_projdir=foam_projdir,
+                    bison_bin=spec['bison'].prefix.bin,
+                    user_dir='$PWD/' + self.build_userdir))
+        set_executable(script)
+
+    def patch(self):
+        """Generate a swakConfiguration file.
+        Provide additional files or other patching.
+        """
+        spec = self.spec
+
+        # Copy files
+        assetdir = join_path(self.package_dir, 'assets')
+        for f in self.assets:
+            install(
+                join_path(assetdir, f),
+                join_path(self.stage.source_path, f))
+
+        # swakConfiguration file
+        config = join_path(self.stage.source_path, self.config_file)
+        with open(config, 'w') as out:
+            if '+python' in spec:
+                out.write(
+                    """# With python
+python_bin={python_bin}
+python_config="$python_bin/python-config"
+export SWAK_PYTHON_INCLUDE="$($python_config --cflags)"
+export SWAK_PYTHON_LINK="$($python_config --ldflags)"
+
+# Force python to be found first
+PATH=$python_bin:$PATH
+#
+""".format(python_bin=spec['python'].prefix.bin))
+            else:
+                out.write('# No python requested\n')
+
+        # Emit openfoam version prior to build
+        # We may be resolved the wrong version, and don't wish to build
+        # it as a dependency
+        tty.info('Build for {0}'.format(spec['openfoam'].format('$_$@$%@+$+')))
```

Comment:
I guess this depends on the size of the underlying package. If I accidentally build the incorrect OpenFOAM version/variant, I've wasted 15-20 min on 24 cpus. Emitting a single line seems to be a small overhead to avoid these types of mistakes.